### PR TITLE
[SPARK-50234][PYTHON][SQL] Improve error message and test for transpose DataFrame API

### PR DESCRIPTION
**Pull request by itholic**
### What changes were proposed in this pull request?

This PR proposes to improve error message and test for transpose DataFrame API

### Why are the changes needed?

To improve error message and negative testing

### Does this PR introduce _any_ user-facing change?

No API changes, but user-facing error message is improved

### How was this patch tested?

Updated the existing UTs

### Was this patch authored or co-authored using generative AI tooling?

No

### CODE REVIEW

**Comment by itholic** [comment]
cc @xinrong-meng

**Review by HyukjinKwon — APPROVED** [review]

File: sql/core/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveTranspose.scala [hunk 1]
```diff
@@ -57,8 +58,8 @@ class ResolveTranspose(sparkSession: SparkSession) extends Rule[LogicalPlan] {
           throw new AnalysisException(
             errorClass = "TRANSPOSE_NO_LEAST_COMMON_TYPE",
             messageParameters = Map(
-              "dt1" -> dt1.sql,
-              "dt2" -> dt2.sql)
+              "dt1" -> toSQLType(dt1.sql),
+              "dt2" -> toSQLType(dt2.sql))
```
Comment (MaxGekk) [inline sql/core/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveTranspose.scala:62]: `toSQLType` over strings is some kind of workaround. In general case, let's leave convertions to strings intirely to `toSQLType`:
```suggestion
              "dt1" -> toSQLType(dt1),
              "dt2" -> toSQLType(dt2))
```

**Review by MaxGekk — CHANGES_REQUESTED** [review]

**Review by xinrong-meng — APPROVED** [review]

**Comment by xinrong-meng** [comment]
LGTM, thank you!

**Comment by itholic** [comment]
Merged to master.
Thanks @HyukjinKwon @MaxGekk @xinrong-meng for the review!

**Pull request closed without merging**